### PR TITLE
Maximal dimension of getLayerPixmap is now configurable

### PR DESCRIPTION
**Pull request by marekhrabe**

### CODE REVIEW

File: lib/generator.js [hunk 1]
```diff
@@ -893,6 +893,9 @@
      *     Note that this option *cannot* be used with an inputRect/outputRect scaling. If inputRect/outputRect
      *     is set, this setting will be ignored and the pixels will not be cropped to document bounds.
      *     (Default: false)
+     * @param {?integer} settings.maxDimension: This is the maximal dimension of pixmap that can be returned
```
Comment (joelrbrandt) [inline lib/generator.js]: The property on the `settings` object is a `number` not an `integer` (there is no integer type in JS). Also, the property is optional, not nullable, so it should be `number=`, not `?number`

File: lib/jsx/getLayerPixmap.jsx [hunk 2]
```diff
@@ -47,8 +47,7 @@
 //         (Default: false)
 //   - compId: number, layer comp ID (optionally and exclusive of compIndex)
 //   - compIndex: number, layer comp index (optionally and exclusive of compId) 
-
-var MAX_DIMENSION = 10000;
+//   - maxDimension: number, maximal dimension of pixmap in pixels
```
Comment (joelrbrandt) [inline lib/jsx/getLayerPixmap.jsx]: You have this parameter in the "optional params" section, but it isn't actually optional (i.e. if someone uses this file separate from the `generator.getLayerPixmap` method, it won't work unless they supply a `params.maxDimension`.

I do agree it should be optional. The correct fix is probably to rename `MAX_DIMENSION` to `DEFAULT_MAX_DIMENSION` and then change line 133 and 134 to something like `params.maxDimension || DEFAULT_MAX_DIMENSION`

**Comment by joelrbrandt** [comment]
Thanks for this! Just a couple of nits, then it's off to mergey mergetown!

File: lib/generator.js (hunk 1 above)
Comment (marekhrabe) [inline lib/generator.js]: the `integer`/`number` choice isn't really clear as there are both variants used in `generator.json`. I changed this to `number` and we could maybe do that for all the other `integers`?

**Comment by iwehrman** [comment]
Any place where we have `@param {integer}` is a bug. We _should be_ following the rules described here: https://developers.google.com/closure/compiler/docs/js-for-compiler Someday, things will be better.

**Comment by joelrbrandt** [comment]
@iwehrman is right, of course. That being said, fixing bugs in old code is a slippery slope. :-) We'll welcome any fixes you want to make, but the primary objective is to make sure new code is great. We shouldn't let great be an enemy of good, as @iwehrman always says.

Anyway, c045257 looks right to me, but I haven't actually tested it. I'll do some testing this weekend and also bump the patch version number and get this baby merged.

*punches one-way ticket to mergey mergetown* :steam_locomotive: :train: :train: :dash:

**Comment by joelrbrandt** [comment]
@marekhrabe Finally had time to test this -- sorry for the delay. Merging!

**Comment by joelrbrandt** [comment]
Updated API wiki page

**Pull request merged**